src/model/custom_electra/electra.py

```diff
@@ -0,0 +1,114 @@
+from typing import Optional
+import torch
+import transformers
+from transformers import ElectraForMaskedLM, ElectraForPreTraining 
+from transformers import AutoConfig, PretrainedConfig, PreTrainedModel, AutoModelForMaskedLM
+#from transformers.tokenization_utils import PretrainedTokenizerFast
+from transformers.models.electra.modeling_electra import ElectraForPreTrainingOutput
+
+
+
+class CustomElectraConfig(PretrainedConfig):
+    model_type= "custom_electra"
+
+    def __init__(
+        self,
+        generator_model: transformers.DebertaV2Config,
+        discriminator_model: transformers.DebertaV2Config,
+        vocab_size: int,
+        embedding_sharing: bool= True,
+        discriminator_lambda_loss: float= 50.0, 
+        pretrained_model_name_or_path= None,
+        **kwargs,
+    ):  
+        print("===== kwargs =====")
+        for key, value in kwargs.items():
+            print(f"{key}: {value}")
+        print(f'pretraineds_model_name_or_path: {pretrained_model_name_or_path}')
+        #super().__init__(pretrained_model_name_or_path)
+        super().__init__(**kwargs)
+        self.vocab_size= vocab_size
+        self.generator_config= generator_model
+        self.discriminator_config= discriminator_model
+        self.embedding_sharing= embedding_sharing
+        self.discriminator_lambda_loss= discriminator_lambda_loss
+        self._name_or_path= pretrained_model_name_or_path
+        
+        #self._name_or_path= CustomElectraConfig 
+# suggest to subclass PreTrainedModel, this has several advantages to using nn.Module
+# you can take advantage of existing functions and ensure it will work with Trainer
+# https://discuss.huggingface.co/t/resources-for-using-custom-models-with-trainer/4151
+class CustomElectraModel(PreTrainedModel):
+
+    # loss weights are default from paper
+    #def __init__(self, config: AutoConfig, tokenizer: PretrainedTokenizerFast=None, lambda_loss=50.0):
+    
+    config_class = CustomElectraConfig
+
+    def __init__(self, config: CustomElectraConfig):
+
+        super().__init__(config)
+
+        self.lambda_loss = self.config.discriminator_lambda_loss
+        self.gumbel_dist = torch.distributions.gumbel.Gumbel(0.,1.)
+
+        # can change this to pass the configs,
+        # or pass the generator/discriminators themselves
+        #base_gen_config = ElectraConfig.from_pretrained('google/electra-base-generator')
+        base_gen_config= AutoConfig.from_pretrained(config.generator_config)
+        self.generator: ElectraForMaskedLM = AutoModelForMaskedLM.from_config(base_gen_config)
+
+        #base_disc_config = ElectraConfig.from_pretrained('google/electra-base-discriminator')
+        base_disc_config = AutoConfig.from_pretrained(config.discriminator_config)
+        self.discriminator: ElectraForPreTraining = AutoModelForMaskedLM.from_config(base_disc_config)
```

Comment:
For the discriminator, we don't want `MaskedLM`, since the task here is replaced token detection.  I'd say there are two options: (i) use `AutoModelForTokenClassification` instead of `AutoModelForMaskedLM`, or (ii) just use `AutoModel`, and add the binary token classification head to this class in `__init__` and `forward`.  Option (i) is probably the smoothest